### PR TITLE
Remove `backtrace` example, move into docs instead

### PR DESCRIPTION
**Pull request by ZedThree**
- Add a new docs page to summarise debugging utilities: `output_debug`, `TRACE`, and the exception backtrace
- Remove the `backtrace` example

### CODE REVIEW

File: manual/sphinx/developer_docs/debugging.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+name of the function it's used in. This is used throughout the main
+library, especially in functions where numerical issues are likely to
+arise.
+
+Backtrace
+=========
+
+Lastly, BOUT++ can also automatically print a backtrace in the event
+of a crash. This is a compile-time option in the BOUT++ library
+(``-DBOUT_ENABLE_BACKTRACE=on``, the default, requires the
```
Comment (dschwoerer) [inline manual/sphinx/developer_docs/debugging.rst:108]:
```suggestion
(``-DBOUT_ENABLE_BACKTRACE=ON``, the default, requires the
```

File: manual/sphinx/developer_docs/debugging.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+``addr2line`` program to be installed), and debug symbols to be turned
+on (``-DCMAKE_BUILD_TYPE=Debug`` or ``=RelWithDebInfo``) in BOUT++
+_and_ the physics model.
```
Comment (dschwoerer) [inline manual/sphinx/developer_docs/debugging.rst:111]:
```suggestion
_and_ the physics model. If debug symbols are only present in part, the
backtrace will be missing names for the other part.
```

File: manual/sphinx/developer_docs/debugging.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    f3() at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:19
+    [bt] #3 ./backtrace() [0x40a3be]
+    f2(int) at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:15
+    [bt] #2 ./backtrace() [0x40a386]
+    f1() at /path/to/BOUT-dev/build/../examples/backtrace/backtrace.cxx:13 (discriminator 2)
+    [bt] #1 ./backtrace(_ZN13BoutExceptionC1IA19_cJEEERKT_DpRKT0_+0xba) [0x40ae16]
+    BoutException::BoutException<char [19]>(char const (&) [19]) at /path/to/BOUT-dev/build/../include/bout/../boutexception.hxx:28 (discriminator 2)
+              
+
+This output tends to be much less user-friendly than the message stack
```
Comment (dschwoerer) [inline manual/sphinx/developer_docs/debugging.rst:142]:
```suggestion
This output tends to be much harder to read than the message stack
```

I find it much friendlier to get in most cases all the info I need without changing the code :-)

File: manual/sphinx/user_docs/physics_models.rst [hunk 4]
```diff
@@ -1110,20 +1095,20 @@ Name                  Useage
...
+command line arguments suppresses the ``output_info`` messages, so
+that they will not appear in the console or log file. Running with
+``-q -q`` suppresses everything except ``output_warn`` and
+``output_error``.
 
 To enable the ``output_debug`` messages, configure BOUT++ with a
 ``CHECK`` level ``>= 3``. To enable it at lower check levels,
-configure BOUT++ with ``-DENABLE_OUTPUT_DEBUG`` (for ``CMake``;
-``--enable-debug-output`` for ``./configure``)). When running BOUT++
+configure BOUT++ with ``-DENABLE_OUTPUT_DEBUG``. When running BOUT++
```
Comment (dschwoerer) [inline manual/sphinx/user_docs/physics_models.rst:1111]:
```suggestion
configure BOUT++ with ``-DBOUT_ENABLE_OUTPUT_DEBUG=ON``. When running BOUT++
```

File: manual/sphinx/user_docs/physics_models.rst [hunk 5]
```diff
@@ -1110,20 +1095,20 @@ Name                  Useage
...
-increases it by one. Running with ``-q`` in the command line arguments
-suppresses the ``output_info`` messages, so that they will not appear
-in the console or log file. Running with ``-q -q`` suppresses
-everything except ``output_warn`` and ``output_error``.
+To reduce the volume of outputs the command line argument ``--quiet``
+(``-q`` for short) reduces the output level by one, and ``--verbose``
+(``-v`` for short) increases it by one. Running with ``-q`` in the
+command line arguments suppresses the ``output_info`` messages, so
+that they will not appear in the console or log file. Running with
+``-q -q`` suppresses everything except ``output_warn`` and
```
Comment (dschwoerer) [inline manual/sphinx/user_docs/physics_models.rst:1106]: ` output_progress` is disabled here, in case someone wonders like me ...

**Review by dschwoerer — CHANGES_REQUESTED** [review]

**Review by bendudson — APPROVED** [review]

**Pull request merged**